rfc/system/1855-type-metadata.md

```diff
@@ -0,0 +1,130 @@
+# [RR] Typing Metadata
+
+**Authors:**
+
+- @maximsmol
+- @kennyworkman
+
+## 1 Executive Summary
+
+A small patch to introduce flytekit support for python-native type metadata.
+
+## 2 Motivation
+
+Users of flyte wish to have access to arbitrary parameter metadata specified in
+typing annotations.
+
+The `LiteralType` definition in flyteidl already has a `metadata` field in its
```

Comment:
Per the discussion in the comments here and on chat, I think it's better to introduce a new message and a corresponding field (and potentially deprecating metadata altogether eventually)...

And the rest of the spec look great. 

```protobuf
message TypeMetadata {
  google.protobuf.Struct annotations = 1;
  // string type_hint = 2;
  // google.protobuf.Struct json_schema = 2;
 ...
}

message LiteralType 
{
...
   TypeMetadata metadata = 6; // For now I guess we have to call it type_metadata
}
```

Reply:
what purpose does replacing  
``` 
google.protobuf.Struct metadata = 6;
```
with a nested message serve. Its just one more level of unwrapping for the exact same information?

Also, doesn't adding an _additional_ field make more sense.?We don't break support for existing use of `metadata` for flyte specific implementation but allows for use of annotations as intended in a way that doesn't conflict.